core/modules/field/field.info.inc

```diff
@@ -614,6 +614,48 @@ function field_info_bundles($entity_type = NULL) {
 }
 
 /**
+ * Returns a lightweight map of fields across bundles.
+ *
+ * The function only returns active, non deleted fields.
+ *
+ * @return
+ *   An array keyed by field name. Each value is an array with two entries:
+ *   - type: The field type.
+ *   - bundles: The bundles in which the field appears, as an array with entity
+ *     types as keys and the array of bundle names as values.
+ * Example:
+ * @code
+ * array(
+ *   'body' => array(
+ *     'bundles' => array(
+ *       'node' => array('page', 'article'),
+ *     ),
+ *     'type' => 'text_with_summary',
+ *   ),
+ * );
+ * @endcode
+ */
+function field_info_field_map() {
+  // Read from the "static" cache.
+  static $field_info_cache = NULL;
```

Comment:
This should use the resettable static pattern. That would look like this:

```
$field_info_cache = backdrop_static(__FUNCTION__, array());
```

Then we'll also need to add a line to `field_info_cache_clear()`:

```
backdrop_static('field_info_field_map');
```

Reply:
added.